server/src/test/java/org/opensearch/search/aggregations/bucket/range/RangeAggregatorTests.java

```diff
@@ -124,6 +132,36 @@ public void testMatchesNumericDocValues() throws IOException {
         });
     }
 
+    public void testTopLevelTermQuery() throws IOException {
+        final String KEYWORD_FIELD_NAME = "route";
+        BooleanQuery.Builder builder = new BooleanQuery.Builder();
+        builder.setMinimumNumberShouldMatch(0);
+        builder.add(new TermQuery(new Term(KEYWORD_FIELD_NAME, "route1")), BooleanClause.Occur.MUST);
+        Query boolQuery = builder.build();
+
+        Document doc1 = new Document();
+        Document doc2 = new Document();
+        Document doc3 = new Document();
+        doc1.add(new NumericDocValuesField(NUMBER_FIELD_NAME, 3));
+        doc2.add(new NumericDocValuesField(NUMBER_FIELD_NAME, 11));
+        doc3.add(new NumericDocValuesField(NUMBER_FIELD_NAME, 12));
+        doc1.add(new KeywordField(KEYWORD_FIELD_NAME, "route1", Field.Store.NO));
+        doc2.add(new KeywordField(KEYWORD_FIELD_NAME, "route1", Field.Store.NO));
+        doc3.add(new KeywordField(KEYWORD_FIELD_NAME, "route2", Field.Store.NO));
+
+        testCase(boolQuery, iw -> {
+            iw.addDocument(doc1);
+            iw.addDocument(doc2);
+            iw.addDocument(doc3);
+        }, range -> {
+            List<? extends InternalRange.Bucket> ranges = range.getBuckets();
+            assertEquals(2, ranges.size());
+            assertEquals(1, ranges.get(0).getDocCount());
+            assertEquals(1, ranges.get(1).getDocCount());
+            assertTrue(AggregationInspectionHelper.hasValue(range));
+        });
+    }
```

Comment:
I would create two versions of this test case. 

One would be the existing one, where you should confirm that the optimization doesn't kick in (because the top-level query doesn't match all docs in the segment).

The other version would create all three docs with `"route1"` as the keyword value. In this case, even though it's not literally a `MatchAllDocsQuery`, the top-level query matches all the docs, so you could apply the optimization.